cognite/neat/rules/models/rules/_dms_architect_rules.py

```diff
@@ -963,43 +1013,47 @@ def _gather_properties(
         return container_properties_by_id, view_properties_by_id
 
     def _create_view_filter(
-        self, view: dm.ViewApply, dms_view: DMSView | None, parent_views: set[dm.ViewId]
-    ) -> dm.Filter:
+        self,
+        view: dm.ViewApply,
+        dms_view: DMSView | None,
+        data_model_type: DataModelType,
+        dms_properties: list[DMSProperty],
+    ) -> DMSFilter:
         selected_filter_name = (dms_view and dms_view.filter_ and dms_view.filter_.name) or ""
         if dms_view and dms_view.filter_ and not dms_view.filter_.is_empty:
-            # Has Explicit Filter
-            return dms_view.filter_.as_filter()
+            # Has Explicit Filter, use it
+            return dms_view.filter_
 
+        if data_model_type == DataModelType.solution and selected_filter_name in [NodeTypeFilter.name, ""]:
+            if (
+                dms_view
+                and isinstance(dms_view.reference, ReferenceEntity)
+                and not dms_properties
+                and (ref_view := self._ref_views_by_id.get(dms_view.reference.as_view_id()))
+                and ref_view.filter
+            ):
+                # No new properties, only reference, reuse the reference filter
+                return DMSFilter.from_dms_filter(ref_view.filter)
+            elif node_ids := {
```

Comment:
**scenario 2** [node type filter]: we go through dms_properties of given dms_view, we check if there is reference stated, if so we create node entity (is this node entity === reference view?, are we to allow multiple node type filters?, as this will create it here?. Then if there are no `node_ids`, alt option is to grab that from actually def of dms_view, if one container `ReferenceEntity`